core/templates/pages/exploration-player-page/layout-directives/learner-view-info.component.spec.ts

```diff
@@ -0,0 +1,194 @@
+import { LearnerExplorationSummaryBackendDict } from 'domain/summary/learner-exploration-summary.model';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { ContextService } from 'services/context.service';
+import { LoggerService } from 'services/contextual/logger.service';
+import { UrlService } from 'services/contextual/url.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { MockTranslatePipe } from 'tests/unit-test-utils';
+import { LearnerViewInfoBackendApiService } from '../services/learner-view-info-backend-api.service';
+import { StatsReportingService } from '../services/stats-reporting.service';
+import { LearnerViewInfoComponent } from './learner-view-info.component';
+
+describe('Learner view info component', () => {
+  let fixture: ComponentFixture<LearnerViewInfoComponent>;
+  let componentInstance: LearnerViewInfoComponent;
+  let contextService: ContextService;
+  let learnerViewInfoBackendApiService: LearnerViewInfoBackendApiService;
+  let loggerService: LoggerService;
+  let readOnlyExplorationBackendApiService:
+    ReadOnlyExplorationBackendApiService;
+  let siteAnalyticsService: SiteAnalyticsService;
+  let statsReportingService: StatsReportingService;
+  let urlInterpolationService: UrlInterpolationService;
+  let urlService: UrlService;
+  let storyViewerBackendApiService: StoryViewerBackendApiService;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      declarations: [
+        LearnerViewInfoComponent,
+        MockTranslatePipe
+      ],
+      providers: [
+        NgbModal,
+        ContextService,
+        LearnerViewInfoBackendApiService,
+        LoggerService,
+        ReadOnlyExplorationBackendApiService,
+        SiteAnalyticsService,
+        StatsReportingService,
+        UrlInterpolationService,
+        UrlService,
+        StoryViewerBackendApiService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(LearnerViewInfoComponent);
+    componentInstance = fixture.componentInstance;
+    contextService = TestBed.inject(ContextService);
+    learnerViewInfoBackendApiService = TestBed.inject(
+      LearnerViewInfoBackendApiService);
+    loggerService = TestBed.inject(LoggerService);
+    readOnlyExplorationBackendApiService = TestBed.inject(
+      ReadOnlyExplorationBackendApiService);
+    siteAnalyticsService = TestBed.inject(SiteAnalyticsService);
+    statsReportingService = TestBed.inject(StatsReportingService);
+    urlInterpolationService = TestBed.inject(UrlInterpolationService);
+    urlService = TestBed.inject(UrlService);
+    storyViewerBackendApiService = TestBed.inject(StoryViewerBackendApiService);
+  });
+
+  afterEach(() => {
+    componentInstance.ngOnDestory();
+  });
+
+  it('should initialize', fakeAsync(() => {
```

Comment:
Could you try to change all the `it` block descriptions to the following format: "should... when..."

Reply:
done

Reply:
Sorry if I wasn't clear. I was expecting 
```diff
- should initialize
+ should initialize when ___ loads or opens
```

Or something along those lines.

Reply:
changed to `should initialize when component loads into view`.